linera-chain/src/data_types.rs

```diff
@@ -767,16 +767,15 @@ impl HashedValue {
         }
     }
 
+    /// Returns the corresponding `ConfirmedBlock`, if this is a `ValidatedBlock`.
```

Comment:
Minor: this might be a surprise upon a first look. I don't think it's worth blocking the PR for, but maybe some alternatives might be:

1. Change the call sites to be something like `(!value.is_confirmed_block()).then(value.into_confirmed_block()).flatten()` (or using an `if` statement)
2. Rename the method to `validated_into_confirmed()`

Reply:
Good idea! Since we want this at all call sites, I'll go with 2.